go.mod: update goavro from 2.9 to 2.10

HonoreDB opened the pull request:
2.10 provides support for large decimals and lets time of day values
be provided as the underlying logical type, which is convenient
because that's how we encode them. It also makes a breaking change
if you were passing such values as duration types, which we were.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67572)
<!-- Reviewable:end -->

HonoreDB committed:
go.mod: update goavro from 2.9 to 2.10

2.10 provides support for large decimals and lets time of day values
be provided as the underlying logical type, which is convenient
because that's how we encode them. It also makes a breaking change
if you were passing such values as duration types, which we were.

Release note: None

stevendanna commented:
Overall this looks good.  I ran `go mod tidy` to clean up some of the issues but still have a vendor conflict to resolve.  I can push a fix for the vendor conflict but am waiting for #68158 to merge before touching it more since we'll just conflict with that as well.

stevendanna commented:
(I have a local branch from before I pushed the tidy if any of this is problematic with work you have locally)

stevendanna reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67572)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @HonoreDB)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/67572#-Mfi3f--1WxJdwXEO3ZP:-Mfi3f--1WxJdwXEO3ZQ:bg7igkv) (no related file):*
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---


<!-- Sent from Reviewable.io -->

HonoreDB commented:
bors r=[stevendanna]
